File: ecommerce-frontend/src/features/auth/services/authService.ts
import api from '@/lib/axios';
import {
  AuthTokens,
  LoginPayload,
  RegisterPayload,
} from '@/types/api.types';

const BASE = '/api/auth';

export const authService = {
  login: (payload: LoginPayload) =>
    api.post<AuthTokens>(`${BASE}/login`, payload).then((r) => r.data),

  register: (payload: RegisterPayload) =>
    api.post<{ email: string; message: string }>(`${BASE}/register`, payload).then((r) => r.data),

  logout: () =>
    api.post<string>(`${BASE}/logout`).then((r) => r.data),

  refreshToken: (refreshToken: string) =>
    api.post<AuthTokens>(`${BASE}/refresh`, { refreshToken }).then((r) => r.data),

  verifyEmail: (email: string, code: string) =>
    api.post<string>(`${BASE}/verify-email?email=${encodeURIComponent(email)}&code=${encodeURIComponent(code)}`).then((r) => r.data),

  resendVerification: (email: string) =>
    api.post<string>(`${BASE}/resend-verification?email=${encodeURIComponent(email)}`).then((r) => r.data),

  changePassword: (oldPassword: string, newPassword: string) =>
    api.put<string>(`${BASE}/password`, { oldPassword, newPassword }).then((r) => r.data),

  forgotPassword: (email: string) =>
    api.post<string>(`${BASE}/forgot-password`, { email }).then((r) => r.data),

  resetPassword: (email: string, otp: string, newPassword: string) =>
    api.post<string>(`${BASE}/reset-password`, { email, otp, newPassword }).then((r) => r.data),
};

File: ecommerce-backend/auth-service/src/main/java/com/ecommerce/auth_service/service/AuthService.java
package com.ecommerce.auth_service.service;

import com.ecommerce.auth_service.dto.request.ChangePasswordRequest;
import com.ecommerce.auth_service.dto.request.ForgotPasswordRequest;
import com.ecommerce.auth_service.dto.request.LoginRequest;
import com.ecommerce.auth_service.dto.request.RefreshTokenRequest;
import com.ecommerce.auth_service.dto.request.RegisterRequest;
import com.ecommerce.auth_service.dto.request.ResetPasswordRequest;
import com.ecommerce.auth_service.dto.response.AuthResponse;

public interface AuthService {
    void register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
    AuthResponse refreshToken(RefreshTokenRequest request);
    void logout(String token);
    void verifyEmail(String email, String code);
    void resendVerificationEmail(String email);
    void forgotPassword(ForgotPasswordRequest request);
    void resetPassword(ResetPasswordRequest request);
    void changePassword(String userId, ChangePasswordRequest request);
}
